ring_doorbell/location.py

```diff
@@ -0,0 +1,125 @@
+# coding: utf-8
+# vim:sw=4:ts=4:et:
+"""Python Ring Location wrapper."""
+import logging
+
+from ring_doorbell.generic import RingGeneric
+from ring_doorbell.const import (
+    APP_URI,
+    LOCATION_MODE_ENDPOINT
+)
+from enum import Enum
+
+_LOGGER = logging.getLogger(__name__)
+
+class LocationMode(Enum):
+    HOME = "home"
+    AWAY = "away"
+
+class RingLocation:
+    """Implementation for Ring Location."""
+
+    def __init__(self, ring, locationData):
+        self._ring = ring
+        self._location_id = locationData["location_id"]
+        self._owner_id = locationData["owner_id"]
+        self._name = locationData["name"]
+        self._latitude = locationData["geo_coordinates"]["latitude"]
+        self._longitude = locationData["geo_coordinates"]["longitude"]
+        self._address1 = locationData["address"]["address1"]
+        self._address2 = locationData["address"]["address2"]
+        self._city = locationData["address"]["city"]
+        self._state = locationData["address"]["state"]
+        self._zip_code = locationData["address"]["zip_code"]
+        self._country = locationData["address"]["country"]
+        self._timezone = locationData["address"]["timezone"]
+        self.update()
```

Comment:
We shouldn’t query in constructor. It should just init the class, no side effects. Also, you just got data?